### ISSUE TITLE
Config file parse errors are extremely unhelpful

### ISSUE THREAD

**Issue by djspiewak** [olafurpg/scalafmt#478]
```
scala.collection.parallel.CompositeThrowable: Multiple exceptions thrown during a parallel computation: org.scalafmt.Error$InvalidScalafmtConfiguration: Unable to parse scalafmt configuration file: .scalafmt.conf
[error] org.scalafmt.cli.Scalafmt210$$anonfun$1.apply(Scalafmt210.scala:22)
[error] org.scalafmt.cli.Scalafmt210$$anonfun$1.apply(Scalafmt210.scala:22)
[error] scala.Option.getOrElse(Option.scala:121)
[error] org.scalafmt.cli.Scalafmt210.format(Scalafmt210.scala:21)
[error] sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[error] sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[error] sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[error] java.lang.reflect.Method.invoke(Method.java:497)
[error] org.scalafmt.sbt.HasScalaFmt.org$scalafmt$sbt$HasScalaFmt$$handleFile(HasScalaFmt.scala:85)
[error] org.scalafmt.sbt.HasScalaFmt$$anonfun$writeFormattedContentsToFiles$1$$anonfun$apply$3.apply(HasScalaFmt.scala:54)
```

That's… informative.

Without line information about the offending error, it's basically impossible to figure out where the problem is.  I'm literally combining trial-and-error with reading `ScalafmtConfig.scala` to try to find the issues.

**Comment by olafurpg** [comment]
Yeah, sorry, the SBT plugin swallows the helpful error messages. For now, I recommend you use the CLI which should print out a helpful error message, for example: https://github.com/olafurpg/scalafmt/blob/master/cli/src/test/scala/org/scalafmt/cli/ConfigTest.scala#L19

The error message is swallowed in SBT exactly here: https://github.com/olafurpg/scalafmt/blob/f59d125a717168077c89f979274676aade20c98d/cli/src/main/scala/org/scalafmt/cli/StyleCache.scala#L28

Here is where the exception is thrown: https://github.com/olafurpg/scalafmt/blob/f59d125a717168077c89f979274676aade20c98d/cli/src/main/scala/org/scalafmt/cli/Scalafmt210.scala#L22-L26

I would greatly appreciate any help with the SBT plugin. I personally use the IntelliJ plugin and CLI so I've put more effort in making the experience nicer with those integrations.

There are `scripted` tests, but you need to run `publishLocal` first to get `scalafmt-cli` in the ivy cache.

**Comment by caoilte** [comment]
We're going to try and look at this in the hackathon.

**Comment by olafurpg** [comment]
Fixed in #478

**Comment by dwijnand** [comment]
Fixed in #482.*
